src/main/scala/midas/passes/ILATopWiring.scala

```diff
@@ -0,0 +1,143 @@
+// See LICENSE for license details.
+package midas
+package passes
+
+import firrtl._
+import firrtl.ir._
+import firrtl.passes._
+import firrtl.annotations._
+
+import java.io._
+import scala.io.Source
+import collection.mutable
+
+import firrtl.transforms.TopWiring._
+
+
+case class ILATopWiringAnnotation(target: ComponentName) extends
+    SingleTargetAnnotation[ComponentName] {
+  def duplicate(n: ComponentName) = this.copy(target = n)
+}
+
+/** Add ports punch out annotated ports out to the toplevel of the circuit.
+    This also has an option to pass a function as a parmeter to generate custom output files as a result of the additional ports
+  * @note This *does* work for deduped modules
+  */
+class ILATopWiringTransform(dir: File = new File("/tmp/")) extends Transform {
+  def inputForm: CircuitForm = LowForm
+  def outputForm: CircuitForm = LowForm
+
+  type InstPath = Seq[String]
+ 
+  def ILAWiringOutputFiles(dir: String, mapping: Seq[((ComponentName, Type, Boolean, InstPath, String), Int)], state: CircuitState): CircuitState = {
+
+    //val targetFile: Option[String] = state.annotations.collectFirst { case ILADebugFileAnnotation(fn) => fn }
+
+    //output vivado tcl file
+    val tclOutputFile = new PrintWriter(new File(dir, "firesim_ila_insert_vivado.tcl" ))
+    //output verilog 'include' file with ports
+    val portsOutputFile = new PrintWriter(new File(dir, "firesim_ila_insert_ports.v" ))
+    //output verilog 'include' file with wires
+    val wiresOutputFile = new PrintWriter(new File(dir, "firesim_ila_insert_wires.v" ))
+    //output verilog 'include' file with ila instantiation
+    val ilaInstOutputFile = new PrintWriter(new File(dir, "firesim_ila_insert_inst.v" ))
+
+    //vivado tcl prologue
+    tclOutputFile.append(s"create_project managed_ip_project $$CL_DIR/ip/firesim_ila_ip/managed_ip_project -part xcvu9p-flgb2104-2-i -ip\n")
+    tclOutputFile.append(s"set_property simulator_language Verilog [current_project]\n")
+    tclOutputFile.append(s"set_property target_simulator XSim [current_project]\n")
+    tclOutputFile.append(s"create_ip -name ila -vendor xilinx.com -library ip -version 6.2 -module_name ila_firesim_0 -dir $$CL_DIR/ip/firesim_ila_ip\n")
```

Comment:
In the spirit of previous comments, maybe the IP version should be pulled out if it needs to be specified / will change.